test/cpp/ext/otel/otel_plugin_test.cc

```diff
@@ -121,6 +121,7 @@ TEST_F(OTelPluginEnd2EndTest, ClientAttemptSentTotalCompressedMessageSize) {
           &data[kMetricName][0].point_data);
   ASSERT_NE(point_data, nullptr);
   ASSERT_EQ(point_data->count_, 1);
+  ASSERT_EQ(absl::get<int64_t>(point_data->max_), 5);
```

Comment:
I don't think this is the right layer to add a test of chttp2 stats collection:
- it's quite disconnected from the code under test, which means when it breaks the author of that pr will have to spend substantial time connecting the dots
- it locks otel usage to chttp2, and it's likely at this point that we'll see otel + chaotic good usage next year
- it ties our testing to otel: if we decide to move to a different telemetry library in the future (this has happened before) we lose test coverage on our transports

I'd recommend instead adding a core end2end test that injects a filter or a custom call tracer, and exercise it just with http2 transports.

Reply:
I can add a http2 specific test (since it'll probably also be useful for future stats stuff), but for these metrics, the transport doesn't matter. All transports should be recording the same data size.

Reply:
We should test the chttp2 framing sizes too though.

Reply:
yeah, added. There might be something interesting going on with the framing bytes though. I want to spend a bit more time over there, so added a TODO for now.